### PR TITLE
[MCPE] Upstream + 1.11

### PR DESCRIPTION
**Pull request by colinrgodsey**
* Upstream merges
* Functional 1.11
* Horse inventory fix
* Tested 1.8, 1.9, 1.10 and 1.11

### CODE REVIEW

File: src/protocolsupport/protocol/packet/middleimpl/clientbound/play/v_pe/CraftingData.java [hunk 1]
```diff
@@ -135,6 +139,9 @@ protected void addRecipeShapeless(ByteBuf to, NetworkItemStack output, List<Netw
 		VarNumberSerializer.writeVarInt(to, 1); // result item count
 		ItemStackSerializer.writeItemStack(to, connection.getVersion(), locale, output);
 		MiscSerializer.writePEUUID(to, UUID.nameUUIDFromBytes(to.array()));
+		if (connection.getVersion().isAfterOrEq(ProtocolVersion.MINECRAFT_PE_1_11)) {
```
Comment (colinrgodsey) [inline src/protocolsupport/protocol/packet/middleimpl/clientbound/play/v_pe/CraftingData.java]: the one and only packet change :)

File: src/protocolsupport/protocol/packet/middleimpl/clientbound/play/v_pe/Map.java [hunk 2]
```diff
@@ -35,6 +36,10 @@ public Map(ConnectionImpl connection) {
 		VarNumberSerializer.writeVarInt(serializer, flags);
 		serializer.writeByte(ChangeDimension.getPeDimensionId(cache.getAttributesCache().getDimension())); //Dimension
 
+		if (version.isAfterOrEq(ProtocolVersion.MINECRAFT_PE_1_11)) {
```
Comment (colinrgodsey) [inline src/protocolsupport/protocol/packet/middleimpl/clientbound/play/v_pe/Map.java]: erm, besides this. 2 packet changes lol

File: resources/resources/pe/blockdefinition.json [hunk 3]
```diff
@@ -4983,4 +3165,20 @@
...
+{"data":4, "id":467, "name":"minecraft:lit_blast_furnace"},
+{"data":5, "id":467, "name":"minecraft:lit_blast_furnace"},
+{"data":6, "id":467, "name":"minecraft:lit_blast_furnace"},
+{"data":7, "id":467, "name":"minecraft:lit_blast_furnace"},
+{"data":0, "id":468, "name":"minecraft:lit_smoker"},
+{"data":1, "id":468, "name":"minecraft:lit_smoker"},
+{"data":2, "id":468, "name":"minecraft:lit_smoker"},
+{"data":3, "id":468, "name":"minecraft:lit_smoker"},
+{"data":4, "id":468, "name":"minecraft:lit_smoker"},
+{"data":5, "id":468, "name":"minecraft:lit_smoker"},
```
Comment (magicus) [inline resources/resources/pe/blockdefinition.json]: There's a lot of changes in block definitions. Are you sure they are correct? Or, rather, do you trust them?

File: src/protocolsupport/protocol/packet/middleimpl/clientbound/play/v_pe/Map.java [hunk 4]
```diff
@@ -25,7 +26,7 @@ public Map(ConnectionImpl connection) {
 	@Override
 	public RecyclableCollection<ClientBoundPacketData> toData() {
 		ClientBoundPacketData serializer = ClientBoundPacketData.create(PEPacketIDs.MAP_ITEM_DATA);
-		VarNumberSerializer.writeSVarLong(serializer, itemData);
+		VarNumberSerializer.writeSVarInt(serializer, itemData);
```
Comment (magicus) [inline src/protocolsupport/protocol/packet/middleimpl/clientbound/play/v_pe/Map.java]: Is this a bug fix, or a change between 1.9/1.10 and 1.11?

Comment (colinrgodsey) [reply src/protocolsupport/protocol/packet/middleimpl/clientbound/play/v_pe/Map.java]: steadfast has a signed var int here, so just switched it over. we only have integer map data anyways, so shouldnt really matter, but this is more correct i think. (can confirm in ghidra once its done doing its migration)

File: src/protocolsupport/protocol/typeremapper/pe/PEDataValues.java [hunk 5]
```diff
@@ -577,9 +579,11 @@ public PEEntityInventoryData init() {
 			private transient NBTCompound filterNBT = null;
 
 			protected void init() {
-				System.out.println("Skipping inventory filter because NBT code still needs to be formatted: " + Filter);
-				//TODO GET THIS???
-				//filterNBT = new createNBTCompoundFromJson(Filter.replaceAll("\'", "\""));
+				try {
```
Comment (colinrgodsey) [inline src/protocolsupport/protocol/typeremapper/pe/PEDataValues.java]: ah right, forgot i added this. this should also fix horse inventories

File: resources/resources/pe/blockdefinition.json (hunk 3 above)
Comment (magicus) [reply resources/resources/pe/blockdefinition.json]: 👍

**Pull request merged**